src/modules/compliance/src/lib/procedures/ensureFilesystemOption.cpp

```diff
@@ -0,0 +1,298 @@
+#include "../Evaluator.h"
+#include <algorithm>
+#include <fstream>
+#include <sstream>
+#include <string>
+#include <vector>
+#include <map>
+#include <iostream>
+#include <set>
+#include <ctime>
+#include <linux/limits.h>
+#include <mntent.h>
+
+namespace compliance {
+struct FstabEntry
+{
+    std::string device;
+    std::string filesystem;
+    std::vector<std::string> options;
+    int dump;
+    int pass;
+    int lineno;
+};
+
+static Result<std::map<std::string, FstabEntry>> ParseFstab(const std::string& filePath)
+{
+    std::map<std::string, FstabEntry> fstabMap;
+    FILE* file = fopen(filePath.c_str(), "r");
+    if (nullptr == file)
+    {
+            return Error("Failed to open file " + filePath + " with error " + strerror(errno));
+    }
+
+    std::string line;
+    int lineno = 0;
+
+    while (!feof(file))
+    {
+        lineno++;
+        char b;
+        fread(&b, 1, 1, file);
+        // Preserve comments and empty lines.
+        if (b == '#' || b == '\n')
+        {
+            while (b != '\n')
+            {
+                fread(&b, 1, 1, file);
+            }
+            continue;
+        } else {
+            fseek(file, -1, SEEK_CUR);
+        }
+        struct mntent *mnt = getmntent(file);
+        std::cout << "MNTENT " << lineno << " " << (void*)mnt << std::endl;
+        if (nullptr == mnt)
+        {
+            continue;
+        }
+        FstabEntry entry;
+        entry.device = mnt->mnt_fsname;
+        entry.filesystem = mnt->mnt_type;
+        entry.dump = mnt->mnt_freq;
+        entry.pass = mnt->mnt_passno;
+        entry.lineno = lineno;
+
+        std::istringstream optionsStream(mnt->mnt_opts);
+        std::string option;
+        while (std::getline(optionsStream, option, ','))
+        {
+            entry.options.push_back(option);
+        }
+
+        fstabMap[mnt->mnt_dir] = entry;
+    }
+    fclose(file);
```

Comment:
ignoring fclose return? Do we care if we cannot close file?

Reply:
It's open for read only, so we don't care (and it should never fail in those circumstances).